src/WebJobs.Script/Binding/HttpBinding.cs

```diff
@@ -261,14 +264,48 @@ public void ProcessResult(IDictionary<string, object> functionArguments, object[
             }
 
             HttpRequestMessage request = (HttpRequestMessage)functionArguments.Values.Union(systemArguments).FirstOrDefault(p => p is HttpRequestMessage);
+            HttpResponseMessage response = null;
+
             if (request != null)
             {
-                HttpResponseMessage response = result as HttpResponseMessage;
-                if (response == null)
+                if (_isProxy)
                 {
-                    response = CreateNegotiatedResponse(request, HttpStatusCode.OK, result);
-                }
+                    var requestObjectFromFunction = result as HttpRequestMessage;
```

Comment:
push all of this proxy specific stuff into a helper method to keep the flow of this method clearer